### PR TITLE
samples: Fix comment typos/errors in pipeline_cache

### PR DESCRIPTION
**Pull request by cdwfs**
This fixes a few minor errors in comment text I found while adapting this code for a forthcoming validation_cache sample, and one very minor potential use-after-free error waiting to happen.

### CODE REVIEW

File: API-Samples/pipeline_cache/pipeline_cache.cpp [hunk 1]
```diff
@@ -299,7 +300,7 @@ int sample_main(int argc, char *argv[]) {
     // End standard draw stuff
 
     if (startCacheData) {
-        // TODO: Create another pipeline, preferably different from the first
+        // TODO: Create another pipeline cache, preferably different from the first
```
Comment (cnorthrop) [inline API-Samples/pipeline_cache/pipeline_cache.cpp]: I agree that we shouldn't have been checking a freed pointer here.  Although with the fix above, this case will never be hit.  Would it be better to move both the free and null assignment to after this check?

Comment (cdwfs) [reply API-Samples/pipeline_cache/pipeline_cache.cpp]: I think it's useful to show that the initial cache data can be safely freed as soon as the cache is created, and would prefer to leave the free/NULL assignment where they are if possible.

I'm not sure why the check against startCacheData is necessary, given the text of the comment it "protects"; the merge function operates on two VkPipelineCache objects, and shouldn't require the initial data at all. How about I leave the TODO comment intact and remove the check that wraps it?

**Comment by cdwfs** [comment]
I checked the job logs from the failed Travis build, but they seem unrelated to this PR:

```
third_party/shaderc/third_party/glslang/hlsl/hlslParseHelper.cpp:70:5: error: 
      field 'cullDistanceInput' will be initialized after field
      'clipDistanceOutput' [-Werror,-Wreorder]
    cullDistanceInput(nullptr),
    ^
1 error generated.
make: *** [obj/local/armeabi-v7a/objs/HLSL/hlsl/hlslParseHelper.o] Error 1
```

File: API-Samples/pipeline_cache/pipeline_cache.cpp (hunk 1 above)
Comment (cnorthrop) [reply API-Samples/pipeline_cache/pipeline_cache.cpp]: Good call.  If you wouldn't mind, remove the if check or make it check against something that makes sense?  I think the intention was to see if we had the initial cache data before bothering to try merging two caches.  There's probably something more accurate to check, but a comment is probably fine.

Comment (cdwfs) [reply API-Samples/pipeline_cache/pipeline_cache.cpp]: Done

**Comment by cnorthrop** [comment]
cherry-picked to master

**Pull request closed without merging**